nvim-types/src/collection.rs

```diff
@@ -55,7 +55,11 @@ impl<T: Clone> Clone for Collection<T> {
 
 impl<T> Drop for Collection<T> {
     fn drop(&mut self) {
-        unsafe { ptr::drop_in_place(self.items.as_ptr()) }
+        unsafe {
+            let slice =
+                std::slice::from_raw_parts_mut(self.items.as_ptr(), self.size);
+            drop(Box::from_raw(slice));
+        }
     }
```

Comment:
Any reason not to do simply `let _ = Vec::from_raw_parts(self.items.as_ptr(), self.size, self.size)`?

Reply:
:shrug: this was first thing that came into mind, because `Collection` is more like a `Box<[T]>` than `Vec<T>`

Reply:
kk, just pushed a small refactor for it. Waiting for CI then we can push it.